webrtc/RTCPeerConnection-helper.js

```diff
@@ -376,96 +308,25 @@ function listenForSSRCs(t, receiver) {
 // It does the heavy lifting of performing signaling handshake,
 // ICE candidate exchange, and waiting for data channel at two
 // end points to open.
-function createDataChannelPair(
-  pc1=new RTCPeerConnection(),
-  pc2=new RTCPeerConnection(),
-  options={})
-{
-  options = Object.assign({}, {
-    channelLabel: '',
-    channelOptions: undefined,
-    doSignaling: true
-  }, options);
-
-  let channel1Options;
-  let channel2Options = null;
-  if (options.channelOptions instanceof Array) {
-    [channel1Options, channel2Options] = options.channelOptions;
-  } else {
-    channel1Options = options.channelOptions;
-  }
-
-  const channel1 = pc1.createDataChannel(options.channelLabel, channel1Options);
-
-  return new Promise((resolve, reject) => {
-    let channel2;
-    let opened1 = false;
-    let opened2 = false;
-
-    function cleanup() {
-      channel1.removeEventListener('open', onOpen1);
-      channel2.removeEventListener('open', onOpen2);
-      channel1.removeEventListener('error', onError);
-      channel2.removeEventListener('error', onError);
-    }
-
-    function onBothOpened() {
-      cleanup();
-      resolve([channel1, channel2]);
+async function createDataChannelPair(
+  pc1 = new RTCPeerConnection(),
+  pc2 = new RTCPeerConnection()) {
+  const pair = [pc1, pc2].map(pc =>
+      pc.createDataChannel('', {negotiated: true, id: 0}));
+  const bothOpen = Promise.all(pair.map(dc => new Promise((r, e) => {
+    dc.onopen = r;
+    dc.onerror = ({error}) => e(error);
+  })));
+  try {
+    exchangeIceCandidates(pc1, pc2);
+    await exchangeOfferAnswer(pc1, pc2);
+    await bothOpen;
+    return pair;
+  } finally {
+    for (const dc of pair) {
+       dc.onopen = dc.onerror = null;
```

Comment:
Clearing `onopen` and `onerror` here should not interfere with any usage outside this helper, since the data channels haven't been exposed outside the helper yet at this point.